programs/storage_api/src/storage_contract.rs

```diff
@@ -45,3 +54,303 @@ pub enum StorageContract {
         reward_validations: Vec<Vec<CheckedProof>>,
     },
 }
+
+pub trait StorageAccount {
```

Comment:
Why the trait here?

Reply:
Where should it be?

Reply:
sorry, I meant, why *a* trait here?

Reply:
Basically, it helps clean up the storage processor. By implementing this trait for KeyedAccount, the process_instruction can trivially invoke the trait on Keyed_account[0]. 

Eventually every program's process_instruction call flow ends up looking really similar and generic (but trait bound obviously) 

Any other reasons I might be missing, @rob-solana?

Reply:
Ok, I see what you're going for. I've taken a similar approach in other places (i.e. KeypairUtil). It's very convenient, but in hindsight, results in some convoluted code that's tedious to extend and makes the underlying data type hard to swap out (i.e. replace "ring" with another "ed25519" library). There's a more straightforward way to do this. Instead of a trait on KeyedAccount, you should "newtype" Account. I'll try it out locally and possibly push a patch here if it's obviously better.

Reply:
@sagar-solana, submitted a PR for this PR here: https://github.com/sagar-solana/solana/pull/2

Reply:
@garious, I see. Doesn't this force a State implementation for each program? Maybe we can #derive that. It was pretty nice and generic with keyed_account too, what specifically was wrong that? 
The way @rob-solana had it structured made the processor code look generic enough that it might be generated one day. Although this doesn't look that hard to auto-gen either. 

I'd like to wait Rob's thoughts on it as well, if that's okay.

Reply:
Good point, State should be on Account, not StoredAccount. Patch on its way!

Reply:
Alright, let me know. I think I'll move ahead with your suggestion once it's ready (and move state to that instruction util file). It'll let me carry on with Replicator work.

Reply:
New patch is up. And it moves State to the util file and deletes the one here and in the Stake instruction processor.